docker/Dockerfile.go-build

```diff
@@ -2,7 +2,7 @@ FROM golang:1.22.4-alpine3.20
 
 WORKDIR /app
 
-RUN apk update && apk add protobuf-dev
+RUN apk update && apk add --no-cache protobuf-dev
```

Comment:
not worth mentioning in the description or commit message, but utilizing --no-cache is a common optimization as due to not caching the apk the container is reduced in size.